File: C#/Monitor.cs
using Virtual;
using System.Drawing;
using System.Reflection.Metadata;
namespace Monitor
{
    class Monitor : Periphery
    {
        private Size Resolution
        { get; set; } = new Size(1920, 1080);
        private int RefreshRate
        { get; set; } = 60;
        private double ScreenSize
        { get; set; } = 23.8;
        public Monitor(
            Size? resolution = null,
            int? refreshRate = null,
            double? screenSize = null,
            string manufacturer = "null")
            : base(manufacturer)
        {
            var res = resolution ?? new Size(1920, 1080);
            if (res.Width < 800 || res.Height < 600 || res.Width > 7680 || res.Height > 4320)
                throw new ArgumentException("Resolution must be between 800x600 and 7680x4320");
            Resolution = res;
            var refresh = refreshRate ?? 60;
            if (refresh < 55 || refresh > 500)
                throw new ArgumentException("Refresh rate must be between 55 and 500 Hz.");
            RefreshRate = refresh;
            var size = screenSize ?? 23.8;
            if (size < 10 || size > 150)
                throw new ArgumentException("The screen diagonal must be between 10 and 150 inches.");
            ScreenSize = size;
        
        }
        public override void Print()
        {
            base.Print();
            Console.WriteLine($"Resolution: {Resolution.Width}x{Resolution.Height}");
            Console.WriteLine($"Refresh Rate: {RefreshRate} Hz");
            Console.WriteLine($"Screen Size: {ScreenSize} inches");
        }

    }
}

File: c++/monitor.cpp
#include "monitor.hpp"
#include <iostream>
#include <stdexcept>

Monitor::Monitor(std::string manufacturer, std::vector<int> resolution, float screenSize, int refreshRate) :
    ComputerTechnology{manufacturer}, Resolution{resolution}, ScreenSize{screenSize}, RefreshRate{refreshRate}
{
    if (resolution.size() != 2 || resolution[0] < 1 || resolution[1] < 1)
        throw std::invalid_argument("Resolution must be a vector of two positive integers");
    if (screenSize <= 0)
        throw std::invalid_argument("Screen size must be a positive float number");
    if (refreshRate < 1)
        throw std::invalid_argument("Refresh rate must be a positive integer");
}

void Monitor::Print() const noexcept
{
    std::cout << "Manufacturer: " << Manufacturer << ", resolution: " << Resolution[0] << 'x' << Resolution[1]
        << ", screen size: " << ScreenSize << "\", refresh rate: " << RefreshRate << " Hz" << std::endl;
}

const std::string& Monitor::GetManufacturer() const noexcept
{
    return Manufacturer;
}

void Monitor::SetManufacturer(std::string manufacturer)
{
    Manufacturer = manufacturer;
}

const std::vector<int>& Monitor::GetResolution() const noexcept
{
    return Resolution;
}

void Monitor::SetResolution(std::vector<int> resolution)
{
    if (resolution.size() != 2 || resolution[0] < 1 || resolution[1] < 1)
        throw std::invalid_argument("Resolution must be a vector of two positive integers");

    Resolution = resolution;
}

void Monitor::SetResolution(int width, int height)
{
    if (width < 1 || height < 1)
        throw std::invalid_argument("Resolution consists of two positive integers");

    Resolution = std::vector<int>({width, height});
}

float Monitor::GetScreenSize() const noexcept
{
    return ScreenSize;
}

void Monitor::SetScreenSize(float screenSize)
{
    if (screenSize <= 0)
        throw std::invalid_argument("Screen size must be a positive float number");

    ScreenSize = screenSize;
}

int Monitor::GetRefreshRate() const noexcept
{
    return RefreshRate;
}

void Monitor::SetRefreshRate(int refreshRate)
{
    if (refreshRate < 1)
        throw std::invalid_argument("Refresh rate must be a positive integer");

    RefreshRate = refreshRate;
}